app/components/point-input.jsx

```diff
@@ -0,0 +1,191 @@
+
+        return () => {
+            autosize.destroy(textareaRef.current)
+        }
+    }, [])
+
+    function getDescWordCount(inputText) {
+        if (!inputText) return 0
+        return inputText.trim().split(/\s+/).length;
+    }
+
+    function getSubjCharCount(inputText) {
+        return inputText.length
+    }
+
+    const isDescValid = inputText => {
+        const wordCount = getDescWordCount(inputText)
+        return wordCount > 0 && wordCount <= maxWordCount
+    }
+
+    const isSubjValid = inputText => {
+        const charCount = getSubjCharCount(inputText)
+        return charCount > 0 && charCount <= maxCharCount
+    }
+
+    const handleSubjectChange = (value) => {
+        setSubject(value)
+        setSubjCharCount(getSubjCharCount(value))
+    }
+
+    const handleDescriptionChange = (value) => {
+        setDescription(value)
+        setDescWordCount(getDescWordCount(value))
+    }
+
+    const handleDescriptionBlur = () => {
+        setIsDescFocused(false)
+        onDone((isSubjValid(subject) && isDescValid(description)), ({ subject: subject, description: description }))
+    }
+
+    return (
+        <>
+            <div className={classes.container}>
+                <input
+                    type="text"
+                    placeholder='Type some thing here'
+                    value={subject}
+                    onChange={(e) => handleSubjectChange(e.target.value)}
+                    onFocus={() => setIsSubjFocused(true)}
+                    onBlur={() => setIsSubjFocused(false)}
+                    className={subjCharCount > maxCharCount ? classes.subject + ' ' + classes.errorInput: classes.subject}>
+
+                </input>
+                {(isSubjFocused || subjCharCount > maxCharCount) && (<span
+                    className={classes.wordCount}
+                >
+                    {subjCharCount} / {maxCharCount}
+                </span>)}
+
+                <textarea
+                    ref={textareaRef}
+                    placeholder="Description"
+                    value={description}
+                    onChange={(e) => handleDescriptionChange(e.target.value)}
+                    onBlur={handleDescriptionBlur}
+                    onFocus={() => setIsDescFocused(true)}
+                    className={descWordCount > maxWordCount ? classes.description + ' ' + classes.errorInput: classes.description}>
+                </textarea>
+                {(isDescFocused || descWordCount > maxWordCount) && (<span
+                    className={classes.wordCount}
+                >{descWordCount} / {maxWordCount}</span>)}
+            </div>
+        </>
+    )
+};
+
+
+const useStyles = createUseStyles(theme => ({
+
+    container: {
+        display: 'flex',
+        flexDirection: 'column',
+        top: '4.0625rem',
+        left: '2.75rem',
+        gap: '0.625rem',
+        width: '100%',
+        height: '13.125rem',
+    },
+    subject: {
+        ...sharedInputStyle(theme),
```

Comment:
This is not wrong, but I want to push back on this pattern. I've seen others use it. Is it an app academy thing?
The css way would be to create another style called sharedInputStyle and then use className=cx(classes.sharedInputStyle, classes.subject)}

But for sharedPlaceHolderStyle the ... pattern looks pretty good.

Reply:
Got it, made the change! Don't think its so much an app academy thing but maybe more of a "I saw this in another component so we're going to mimic it" type of thing.

I guess when should we use the ... pattern vs creating another style and using cx()? Like for sharedHoverStyle, sharedErrorStyle, and sharedWordCountStyle is ... pattern ok, or should it only be used for pseudo classes/elements?